crates/uv-client/src/base_client.rs

```diff
@@ -136,6 +137,13 @@ impl<'a> BaseClientBuilder<'a> {
 
         // Initialize the base client.
         let client = self.client.clone().unwrap_or_else(|| {
+            if CryptoProvider::get_default().is_none() {
+                // Use aws_lc_rs as the default TLS provider to support more SSL cert algorithms
+                aws_lc_rs::default_provider()
+                    .install_default()
+                    .expect("failed to install aws_lc_rs as default TLS provider");
```

Comment:
Is this always safe due to the above `if`?

Reply:
Yes I think so, but when I tested removing this `expect()` I saw that if `install_default()` did fail then no error would be reported and the TLS would fall back to `ring` which I wasn't sure was desired. Is there a different mechanism than `expect()` we should use here to avoid that?

Reply:
There are a couple options, like ignoring the result:

```rust
let _ = aws_lc_rs::default_provider().install_default();`
```

or warning the user on failure:

```rust
if let Err(err) = aws_lc_rs::default_provider().install_default() {
    warn_user_once!(...)
}
```

`expect` is okay if it should _never_ fail like... an assertion of an invariant. I don't understand when this would fail though, I'd need to poke around its implementation / documentation.